src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -1394,22 +1380,24 @@ class G1MergeHeapRootsTask : public WorkerTask {
       {
         // 2. collection set
         G1MergeCardSetClosure merge(_scan_state);
-        G1ClearBitmapClosure clear(g1h);
-        G1CombinedClosure combined(&merge, &clear);
 
         if (_initial_evacuation) {
           G1HeapRegionRemSet::iterate_for_merge(g1h->young_regions_cardset(), merge);
         }
 
-        g1h->collection_set_iterate_increment_from(&combined, nullptr, worker_id);
-        G1MergeCardSetStats stats = merge.stats();
-
+        g1h->collection_set_iterate_increment_from(&merge, worker_id);
         for (uint i = 0; i < G1GCPhaseTimes::MergeRSContainersSentinel; i++) {
-          p->record_or_add_thread_work_item(merge_remset_phase, worker_id, stats.merged(i), i);
+          p->record_or_add_thread_work_item(merge_remset_phase, worker_id, merge.stats().merged(i), i);
```

Comment:
`stats()` has side-effect; should be invoked only once.

Reply:
Nice catch! Fixed.